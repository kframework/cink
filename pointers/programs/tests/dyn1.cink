int main() {
  int *p = new int;
  bool *q;
  q = new bool;
  *q = true;
  if (*q) *p = 7;
  cout << *p << endl;
  delete p;
  delete q;
}

