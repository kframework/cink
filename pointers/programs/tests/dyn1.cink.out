7
<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            noVal
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            main |-> #symScalLoc(1)
        </env>
    </control>
    <genv>
        main |-> #symScalLoc(1)
    </genv>
    <store>
        #symScalLoc(1) |-> lambda (  @ p of pointer to int ; (p = (new int
          )) ; q of pointer to bool ; (q = (new bool)) ; ((* q) = true) ; if (
           (* q) ) ((* p) = 7) ; else { } ((cout << (* p)) << (endl)) ; (
          delete p) ; (delete q) ; )
        #symScalLoc(3) |-> 7
        #symScalLoc(5) |-> true
    </store>
</T>
